js/main.js

```diff
@@ -75,33 +96,70 @@ function stayOnSamePageAfterRefresh() {
 stayOnSamePageAfterRefresh();
 
 function renderEntries(entry) {
+
   var $entryRow = document.createElement('ROW');
   var $imgDiv = document.createElement('DIV');
   var $entryImg = document.createElement('IMG');
   var $textDiv = document.createElement('DIV');
+  var $h3AndEditRow = document.createElement('ROW');
+  var $titleDiv = document.createElement('DIV');
   var $entryTitle = document.createElement('H3');
+  var $iconDiv = document.createElement('DIV');
+  var $editIcon = document.createElement('I');
   var $entryText = document.createElement('P');
 
   $entryRow.className = 'row dom-row-layout';
   $imgDiv.className = 'column-half';
   $textDiv.className = 'column-half';
+  $h3AndEditRow.className = 'row title-edit-layout';
+  $titleDiv.className = 'title-div-layout';
+  $editIcon.className = 'fas fa-pen';
   $entryTitle.className = 'title-margin-top';
 
   $entryImg.setAttribute('src', entry.photoURL);
+  $entryRow.setAttribute('data-entry-id', currentEntryId);
   $entryTitle.textContent = entry.title;
   $entryText.textContent = entry.message;
 
   $entryRow.appendChild($imgDiv);
-  $entryRow.appendChild($textDiv);
   $imgDiv.appendChild($entryImg);
-  $textDiv.appendChild($entryTitle);
+  $entryRow.appendChild($textDiv);
+  $textDiv.appendChild($h3AndEditRow);
+  $h3AndEditRow.appendChild($titleDiv);
+  $h3AndEditRow.appendChild($iconDiv);
+  $titleDiv.appendChild($entryTitle);
+  $iconDiv.appendChild($editIcon);
   $textDiv.appendChild($entryText);
+
+  $entryRow.addEventListener('click', handleEdit);
+
+  function handleEdit(event) {
+    // debugger;
+    if (event.target !== $editIcon) {
+      return;
+    }
+    showEntryForm();
+    var entriesParent = event.target.closest('.dom-row-layout');
+    var entriesParentId = entriesParent.getAttribute('data-entry-id') * 1;
+
+    for (var i = 0; i < data.entries.length; i++) {
+      if (entriesParentId === data.entries[i].nextEntryId - 1) {
+        data.editing = entriesParent;
+        $image.src = data.entries[i].photoURL;
+        $title.value = data.entries[i].title;
+        $photoURL.value = data.entries[i].photoURL;
+        $message.value = data.entries[i].message;
+        break;
+      }
+    }
```

Comment:
Instead of listening to every single edit icon, use event delegation by listening to the list element and checking if the event target was an edit icon.